CMSgov/managed-care-review
# MR-2886 CMS can upload a Q&A document

## Summary
Add file upload to add questions page. 

Other refactors: 
-  new React hooks -  `useFileUpload` and `useErrorSummary`. 
- edited `useS3` to include handling for bucket name and also hold S3 utility functions

#### Related issues
https://qmacbis.atlassian.net/browse/MR-2886 
Also fixed this viewport scroll position bug https://qmacbis.atlassian.net/browse/MR-2997 

#### Screenshots
![Screenshot 2023-02-23 at 11 33 56 AM](https://user-images.githubusercontent.com/10750442/220985196-7ceb4e98-a947-4f11-bbe3-db347fb40aa7.png)

#### Test cases covered


JEST:
- file upload accepts multiple pdf, word, excel documents
- displays form validation error if attempting to add question with zero files
- displays file upload alert if attempting to add question with all invalid files
- displays api error if createQuestion fails

CYPRESS:
- none yet, will address this as part of https://qmacbis.atlassian.net/browse/MR-2887 so we just add one full test for Q&A.


## QA guidance
See ticket AC and also
- Make sure file uploads, downloads, bulk downloads still working (since I changed how s3 works slightly to handle different buckets for different use cases)
